projects/packages/jetpack-mu-wpcom/src/features/launchpad/launchpad-task-definitions.php

```diff
@@ -518,6 +528,37 @@ function wpcom_track_publish_first_post_task() {
 	wpcom_mark_launchpad_task_complete_if_active( 'first_post_published_newsletter' );
 }
 
+/**
+ * Callback for completing the Write posts task.
+ *
+ * @return void
+ */
+function wpcom_track_write_posts_task() {
+	// Ensure that Headstart posts don't mark this as complete
+	if ( defined( 'HEADSTART' ) && HEADSTART ) {
+		return;
+	}
+
+	$posts_count = wp_count_posts( 'post' );
+	if ( $posts_count->publish >= 3 ) {
+		wpcom_mark_launchpad_task_complete_if_active( 'write_posts' );
+	}
```

Comment:
I think we need to ensure that we're excluding posts that were previously created by Headstart. I think we can do that as follows:
```suggestion
	$total_posts_count = wp_count_posts( 'post' );
	
	// Make sure we don't count any published posts that were created by Headstart.
	$headstart_posts_filter = array(
		'post_type' => 'post',
		'fields' => 'ids',
		'status' => 'publish',
		'meta_key' => '_hs_old_id',
		'meta_compare' => 'EXISTS',
	);
	$headstart_post_count = count( get_posts( $headstart_posts_filter ) );

	if ( ( $posts_count->publish - $headstart_posts_count ) >= 3 ) {
		wpcom_mark_launchpad_task_complete_if_active( 'write_posts' );
	}
```